### PR TITLE
Mib 432 invalid value

### PR DESCRIPTION
**Pull request by PasGat**

### CODE REVIEW

File: test/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberControllerSpec.scala [hunk 1]
```diff
@@ -40,11 +40,12 @@ class EoriNumberControllerSpec extends DeclarationJourneyControllerSpec {
 
     val result = controller.onSubmit()(
       buildPost(routes.EoriNumberController.onSubmit().url, aSessionId)
-        .withFormUrlEncodedBody(("eori", "GB123467800000"))
+        .withFormUrlEncodedBody(("eori", "GB123467800022"))
     )
 
     status(result) mustBe 400
     contentAsString(result) must include(messages("eoriNumber.error.notFound"))
+    contentAsString(result) must include(messages("GB123467800022"))
```
Comment (jordanrowe) [inline test/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberControllerSpec.scala:48]: don't need messages here?

Comment (PasGat) [reply test/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberControllerSpec.scala:48]: good catch

File: app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala [hunk 2]
```diff
@@ -72,18 +71,16 @@ class EoriNumberController @Inject()(
...
+      Future.successful(badRequestResult(isAgent, declarationType, eori))
 
-  private def badRequestResult(isAgent: YesNo, declarationType: DeclarationType)(
-    implicit request: DeclarationJourneyRequest[AnyContent]): Result = {
-    val formWithError = form(isAgent, request.declarationType).withError(FormError("eori", "eoriNumber.error.notFound"))
-    BadRequest(view(formWithError, isAgent, backButtonUrl, declarationType))
-  }
+  private def badRequestResult(isAgent: YesNo, declarationType: DeclarationType, eori: String)(
+    implicit request: DeclarationJourneyRequest[AnyContent]): Result =
+    BadRequest(view(formWithError(isAgent, request.declarationType, eori), isAgent, backButtonUrl, declarationType))
```
Comment (jordanrowe) [inline app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala:85]: I think you've gone the long way around about this, does formWithError not have a .fill() method

so would just be BadRequest(view(formWithError(isAgent, request.declarationType).fill(eori)...

Comment (jordanrowe) [reply app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala:85]: or actually even better, when you declare your val formWithError just stick a .fill() after the .withError()

Comment (PasGat) [reply app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala:85]: you mean 
`form(customsAgent, declarationType)
      .withError(FormError(fieldName, "eoriNumber.error.notFound"))
      .bind(Map(fieldName -> eori))`
Vs
`form(customsAgent, declarationType)
      .withError(FormError(fieldName, "eoriNumber.error.notFound"))
      .fill(eori)`
???

Comment (jordanrowe) [reply app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala:85]: If I understand correctly, what you are trying to do is prepopulate the field with their EORI on the page

So you don't want to call bind, that will bind the form and return a String?

You instead want an instance of the form, with an error, but the form is filled with their previous submission

**Review by jordanrowe — APPROVED** [review]

**Pull request merged**